src-plugins/itkProcessRegistrationDiffeomorphicDemons/itkProcessRegistrationDiffeomorphicDemons.cpp

```diff
@@ -117,15 +98,25 @@ QString itkProcessRegistrationDiffeomorphicDemons::identifier() const
 
 
 template <typename PixelType>
-        int itkProcessRegistrationDiffeomorphicDemonsPrivate::update()
+int itkProcessRegistrationDiffeomorphicDemonsPrivate::update()
 {
     typedef itk::Image< PixelType, 3 >  FixedImageType;
     typedef itk::Image< PixelType, 3 >  MovingImageType;
 
-    //unfortunately diffeomorphic demons only work with double or float types...
-    // so we need to use a cast filter.
     typedef itk::Image< float, 3 > RegImageType;
     typedef double TransformScalarType;
+
+    // Check that the inputs are the same size/origin/spacing
+    if ( (proc->fixedImage()->GetLargestPossibleRegion().GetSize()
+          != proc->movingImages()[0]->GetLargestPossibleRegion().GetSize())
+         || (proc->fixedImage()->GetOrigin()
+             != proc->movingImages()[0]->GetOrigin())
+         || (proc->fixedImage()->GetSpacing()
+             != proc->movingImages()[0]->GetSpacing()) )
+    {
+        return medAbstractProcess::MISMATCHED_DATA_SIZES_ORIGIN_SPACING;
```

Comment:
No, my question still makes sense :) Why do you want the inputs the same size/... ?